contracts/LockDealNFT/LockDealNFTState.sol

```diff
@@ -34,6 +34,27 @@ abstract contract LockDealNFTState is ERC721Enumerable, ILockDealNFTEvents, Owna
         }
     }
 
+    function getUserDataByTokens(
+        address user,
+        address[] memory tokens,
+        uint256 from,
+        uint256 to
+    ) public view returns (BasePoolInfo[] memory userPoolInfo) {
+        require(from <= to, "Invalid range");
+        require(to - from < balanceOf(user),"Range greater than user pool count");
+        userPoolInfo = new BasePoolInfo[](to - from + 1);
+        uint256 userPoolCount = 0;
+        for (uint256 i = from; i <= to; ++i) {
+            uint256 poolId = tokenOfOwnerByIndex(user, i);
+            for (uint256 j = 0; j < tokens.length; ++j) {
+                if (tokens[j] == tokenOf(poolId)) {
```

Comment:
I think we got this in helpers. is it in an array?

Reply:
fixed
https://github.com/The-Poolz/LockDealNFT/blob/91435793c9e40f05e93ba0d535b25b92d25bb15b/contracts/LockDealNFT/LockDealNFTState.sol#L50-L52